### PR TITLE
cmake: make check

### PR DESCRIPTION
**Pull request by alimaredia**
The README.md can be left out and edited later.

Running the following commands caused all the unittests to build and run. They weren't all passing yet, but thats a discussion for a little later.
(assuming "git submodule update --init --recursive" has been run)
$ cmake ../path/to/ceph/
$ make check

### CODE REVIEW

**Comment by tchaikov** [comment]
@alimaredia seems your pull request does not merge, could you rebase it against master, and repush?

**Comment by alimaredia** [comment]
Kefu,

I added the changes you asked for, thanks for the pointers. This is my first ever pull request!

-Ali

**Comment by tchaikov** [comment]
@alimaredia you need to add "Signed-off-by" to all your commits. now only 93fda8e has it. and it's good to have a prefix on your pull request's titile. but i was suggesting to add the prefix to all your commit messages. we always add a prefix to the title of the commit message, if it's revise of documentation, it's likely to start with "doc: ". if the change is related to monitor, it's started with "mon: ", etc. so other developers can find interesting commits when looking at the "git log", for example. does it make sense to you?

**Comment by alimaredia** [comment]
The unittests in erasure code were reordered because some of the tests required extra "-I/path" compile flags. By using the include_directories, all the targets after an include_directories command use the paths included. 

By putting the targets that needed the extra include paths at the end, I ensured there were no extra flags on targets that didn't require them in Makefile.am.

At the end of the day, it shouldn't make a difference, and I can understand why you would want to keep them in the old more logically progressive order.

File: ceph-object-corpus [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 20351c6bae6dd4802936a5a9fd76e41b8ce2bad0
+Subproject commit 7bd9bbe235c59527dd1ab192fed21944da686d20
```
Comment (tchaikov) [inline ceph-object-corpus]: @alimaredia could you remove this change if it is not intended?

**Comment by tchaikov** [comment]
@alimaredia ping?

**Comment by dachary** [comment]
@alimaredia I'm around if you need to discuss issues with erasure code tests (because they hardcode .lib)

**Pull request merged**